mindsdb/integrations/handlers/statsforecast_handler/statsforecast_handler.py

```diff
@@ -133,14 +141,20 @@ def predict(self, df, args={}):
         sf = StatsForecast(models=[], freq=model_args["frequency"], df=training_df)
         sf.fitted_ = fitted_models
         forecast_df = sf.predict(model_args["horizon"])
-        forecast_df = forecast_df[forecast_df.index.isin(groups_to_keep)]
-        return get_results_from_nixtla_df(forecast_df, model_args)
+        results_df = forecast_df[forecast_df.index.isin(groups_to_keep)]
+
+        if model_args["hierarchy"]:
+            hier_df = dill.loads(self.model_storage.file_get("hier_df"))
+            hier_dict = dill.loads(self.model_storage.file_get("hier_dict"))
+            reconciled_df = reconcile_forecasts(training_df, forecast_df, hier_df, hier_dict)
+            results_df = reconciled_df[reconciled_df.index.isin(groups_to_keep)]
+        return get_results_from_nixtla_df(results_df, model_args)
 
     def describe(self, attribute=None):
         model_args = self.model_storage.json_get("model_args")
 
         if attribute == "model":
-            return pd.DataFrame({k: [model_args[k]] for k in ["model_name", "frequency", "season_length"]})
+            return pd.DataFrame({k: [model_args[k]] for k in ["model_name", "frequency", "season_length", "hierarchy"]})
```

Comment:
Wouldn't this break on non-hierarchical models (i.e. `KeyError "hierarchy" key not found` or some such)?

Reply:
Nevermind, this is covered in other tests and the CI is passing ✅